furusiya/model/maps/area_map.py

```diff
@@ -57,15 +58,31 @@ def place_on_random_ground(self, entity):
         entity.y = y
         self.entities.append(entity) 
 
+def filter_tiles(tiles, filter_callback):
+    return [
+        (x, y)
+        for x, y in tiles
+        if filter_callback(x, y)
+    ]
 
-class MapTile:
-    def __init__(self, walkable=False, block_sight=None):
-        self.is_walkable = walkable
-        self.character = '.'
-        self.colour = (128, 128, 128)
-        self.dark_colour = (32, 32, 32)
 
+# TODO: extract into its own file
+class MapTile:
+    def __init__(self, walkable=False):
         self.is_explored = False
+        self.convert_to_ground()
 
-        # by default, if a tile is blocked, it also blocks sight
-        self.block_sight = block_sight or not self.is_walkable
+    def convert_to_wall(self, character=AreaMap.WALL_CHARACTER, colour=AreaMap.LIGHT_WALL_COLOUR, dark_colour=AreaMap.DARK_WALL_COLOUR):
```

Comment:
I seriously don't like this using constants from another class. Perhaps they belong in here instead?

Reply:
I don't know. I think anything to do with display (character, colour) should be ripped out of this "model" class. A common pattern I've seen is MV (ModelView) classes, eg. a `TileView` class that has a `MapTile` and a character/colour. Let's think on this.